src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java

```diff
@@ -112,10 +90,41 @@ public void addComparisonExpressions(List<ComparisonExpression> comparisonExpres
     }
   }
 
+  /**
+   * Add childLogicalExpression after it is been flipped with $not operator Will not increment the
+   * counter to avoid double count
+   *
+   * @param logicalExpression logicalExpression
+   */
+  public void addLogicalExpressionFlipped(LogicalExpression logicalExpression) {
+    // skip empty logic expression
+    if (logicalExpression.isEmpty()) {
+      return;
+    }
+    logicalExpressions.add(logicalExpression);
+  }
+
+  /**
+   * Add childComparisonExpressions after they have been flipped with $not operator Will not
+   * increment the counter to avoid double count
+   *
+   * @param comparisonExpressionList comparisonExpressionList
+   */
+  public void addComparisonExpressionsFlipped(List<ComparisonExpression> comparisonExpressionList) {
+    if (comparisonExpressionList.isEmpty()) {
+      return;
+    }
+    comparisonExpressions.addAll(comparisonExpressionList);
+  }
+
   public LogicalOperator getLogicalRelation() {
     return logicalRelation;
   }
 
+  public void setLogicalRelation(LogicalOperator logicalRelation) {
```

Comment:
Add java doc, it's called only from FilterClause, make access modifier to `protected`

Reply:
solved